projects/FW_102/src/main.c

```diff
@@ -0,0 +1,57 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "delay.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void inc_counter_a(SoftTimerId timer_id, void *context);
+static void inc_counter_b(SoftTimerId timer_id, void *context);
+
+int main(void) {
+  // Initialize soft timer, and soft timer dependency (interrupt is needed)
+  interrupt_init();
+  soft_timer_init();
+
+  Counters loop_counter = {
+    .counter_a = 0,
+    .counter_b = 0,
+  };
+
+  // Increment counter a every 0.5s, counter b ever 1s.
+  soft_timer_start_millis(500, inc_counter_a, &loop_counter, 0);
+  soft_timer_start_millis(1000, inc_counter_b, &loop_counter, 0);
+
+  // Wait for an interrupt
+  while (true) {
+    wait();
```

Comment:
Awesome job using `wait()`!

Reply:
Thank you! For reviews, would you like me to respond to the comments that I have made changes on, or should I let the `outdated` tag speak for itself?

Reply:
If you have questions / comments then definitely comment, otherwise the `outdated` tag is fine; resolving conversations for comments you've resolved is also good.

Reply:
Should I let the reviewer resolve it as a standard practice, or is this more of a flexible rule dependent on the reviewer?

Reply:
Eh it's not too strict, I know I appreciate it when people resolve my comments :)